Cabal/Cabal.cabal

```diff
@@ -315,6 +315,7 @@ library
     Distribution.Compat.Exception
     Distribution.Compat.Graph
     Distribution.Compat.Internal.TempFile
+    Distribution.Compat.Prelude
```

Comment:
@ezyang one could also use a module-level warning (in the exposed `.Internal` module; Cabal's own code would still use the non-.Internal one) so people really notice the prominent comment